scripts/create_test_subset.py

```diff
@@ -390,6 +437,63 @@ def get_map_ipid_esid(
     return external_sample_internal_participant_map
 
 
+def get_samples_for_families(project: str, additional_fams: Optional[List[str]] = None):
+    """Returns the samples that belong to a list of families"""
+
+    samples: List[str] = []
+    full_pedigree = fapi.get_pedigree(
+        project=project,
+        replace_with_participant_external_ids=False,
+        replace_with_family_external_ids=True,
+    )
+
+    ipids = [
+        family['individual_id']
+        for family in full_pedigree
+        if family['family_id'] in additional_fams
+    ]
+
+    sample_objects = sapi.get_samples(
+        body_get_samples={
+            'project_ids': [project],
+            'participant_ids': ipids,
+            'active': True,
+        }
+    )
+
+    samples = [sample['id'] for sample in sample_objects]
+
+    return samples
+
+
+def get_fams_for_samples(
+    project: str,
+    additional_samples: Optional[List[str]] = None,
+):
+    """Returns the families that a list of samples belong to"""
+    fams: List[str] = []
+    sample_objects = sapi.get_samples(
+        body_get_samples={
+            'project_ids': [project],
+            'sample_ids': additional_samples,
+            'active': True,
+        }
+    )
+
+    pids = [sample['participant_id'] for sample in sample_objects]
+    full_pedigree = fapi.get_pedigree(
+        project=project,
+        replace_with_participant_external_ids=False,
+        replace_with_family_external_ids=True,
+    )
+
+    fams = [
+        fam['family_id'] for fam in full_pedigree if str(fam['individual_id']) in pids
```

Comment:
I think this might result in duplicate family IDs as a family Id would appear multiple times in a pedigree. Good to `set` and cast to list if needed as a list (otherwise a Set in this usage makes sense IMO).